### ISSUE TITLE
Model.trainable=False does not work

### ISSUE THREAD

**Issue by LianShuaiLong** [keras-team/keras#14325]
I want to finetune Efficientnet B7 model with Keras, so i try to freeze conv layers of Efficientnet B7 and just train FC layer.The following is my code:
   
 x = keras.layers.GlobalAveragePooling2D()(conv_base.output)
    output = keras.layers.Dense(2, activation='softmax')(x)
    model = keras.models.Model(inputs=[conv_base.input], outputs=[output])
    print('This is the number of trainable layers before freezing the conv base:',len(model.trainable_weights))
    conv_base.trainable = False
    model.compile(loss='categorical_crossentropy',
                  optimizer='SGD',
                  metrics=['accuracy']
                  )
    print('This is the number of trainable layers after freezing the conv base:',len(model.trainable_weights))

but i just find out Model.trainable=False does not work as the log shows:

This is the number of trainable layers before freezing the conv base: 711
This is the number of trainable layers after freezing the conv base: 711

**Comment by sushreebarsa** [comment]
Could you please refer to this [documentation ](https://keras.io/examples/vision/image_classification_efficientnet_fine_tuning/) and let us know if it helps ? Thanks !

**Comment by sushreebarsa** [comment]
Closing this issue due to lack of recent activity & assuming this issue as resolved .Please feel free to reopen if you still have concern.Thanks!
